### ISSUE TITLE
`json/encode` and representing objects with `null` values

### ISSUE THREAD

**Issue by sogaiu** [janet-lang/spork#159]
It appears that `json/encode` can produce buffers that have `null` in them in some situations:

```
(json/encode nil)
# =>
@"null"
```

```
(json/encode [1 2 nil 9])
# =>
@"[1,2,null,9]"
```

However, it's not clear to me whether one can produce something that's supposed to represent an object (note that the examples above involve a string and an array) with a `null` value in it.  That is, something like:

```
@`{"result":null}`
```

and not:

```
@`{"result":"null"}`
```

At some point, CFiggers' janet-lsp fork had a [work-around for this situation](https://github.com/CFiggers/janet-lsp/blob/8b3b56d305c95dc1bde8a0cd01f6b2a0dce50fdb/src/rpc.janet):

```janet
(defn success-response [id result]
  (if (nil? result)
    (string "{\"id\":" id ",\"result\":null,\"jsonrpc\":\"2.0\"}")
    (json/encode {:jsonrpc "2.0"
                  :id id
                  :result result})))
```

For reference, here is a [link](https://janet.zulipchat.com/#narrow/stream/410074-spork/topic/null.20and.20json.2Fencode/near/397676636) to the beginning of a related discussion on Zulip.

**Comment by bakpakin** [comment]
The function `json/decode` has an optional parameter `nils` which lets you change the behavior with how to encode null. The default is `:null`, which will not cause a loss of information, but you can pass in a truthy value to get a normal `nil`.

From `(doc json/decode)`:

```
cfunction

    (json/decode json-source &opt keywords nils)
    
    Returns a janet object after parsing JSON. If keywords is truthy, string keys will be
    converted to keywords. If nils is truthy, null will become nil instead of the keyword :null.
```

**Comment by sogaiu** [comment]
I'm confused.

How does the capability of `json/decode` help with respect to `json/encode`?

IIUC, `json/encode` is able to produce strings representing some types of JSON values that have `null` in them, but it's not clear how to do this for JSON objects.

For JSON strings and JSON arrays:

**Comment by bakpakin** [comment]
Should be fixed in d644da0fd05612a2d5a3c97277bf7b9bb96dcf6b - now passing `:null` to `json/encode` will result in a `null` in the JSON.

**Comment by sogaiu** [comment]
Seems to work:

```
(json/encode {:result :null})
# =>
@"{\"result\":null}"
```

Thank you!
